File: app/src/main/java/com/mxt/anitrend/adapter/pager/index/SeasonPageAdapter.java
package com.mxt.anitrend.adapter.pager.index;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;

import com.mxt.anitrend.R;
import com.mxt.anitrend.base.custom.pager.BaseStatePageAdapter;
import com.mxt.anitrend.util.GraphUtil;
import com.mxt.anitrend.util.KeyUtil;
import com.mxt.anitrend.view.fragment.list.MediaBrowseFragment;

/**
 * Created by Maxwell on 10/14/2016.
 */
public class SeasonPageAdapter extends BaseStatePageAdapter {

    public SeasonPageAdapter(FragmentManager manager, Context context) {
        super(manager, context);
        setPagerTitles(R.array.seasons_titles);
    }

    /**
     * Return the Fragment associated with a specified position.
     *
     * @param position
     */
    @Override
    public Fragment getItem(int position) {
        switch (position)
        {
            case 0:
                return MediaBrowseFragment.newInstance(getParams(), GraphUtil.getDefaultQuery(true)
                        .putVariable(KeyUtil.arg_mediaType, KeyUtil.ANIME)
                        .putVariable(KeyUtil.arg_season, KeyUtil.WINTER));
            case 1:
                return MediaBrowseFragment.newInstance(getParams(), GraphUtil.getDefaultQuery(true)
                        .putVariable(KeyUtil.arg_mediaType, KeyUtil.ANIME)
                        .putVariable(KeyUtil.arg_season, KeyUtil.SPRING));
            case 2:
                return MediaBrowseFragment.newInstance(getParams(), GraphUtil.getDefaultQuery(true)
                        .putVariable(KeyUtil.arg_mediaType, KeyUtil.ANIME)
                        .putVariable(KeyUtil.arg_season, KeyUtil.SUMMER));
            case 3:
                return MediaBrowseFragment.newInstance(getParams(), GraphUtil.getDefaultQuery(true)
                        .putVariable(KeyUtil.arg_mediaType, KeyUtil.ANIME)
                        .putVariable(KeyUtil.arg_season, KeyUtil.FALL));
        }
        return null;
    }
}

File: app/src/main/java/com/mxt/anitrend/adapter/pager/index/SeasonPageAdapter.kt
package com.mxt.anitrend.adapter.pager.index

import android.content.Context
import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import com.mxt.anitrend.R
import com.mxt.anitrend.base.custom.pager.BaseStatePageAdapter
import com.mxt.anitrend.extension.LAZY_MODE_UNSAFE
import com.mxt.anitrend.ui.fragmentByTagOrNew
import com.mxt.anitrend.ui.model.FragmentItem
import com.mxt.anitrend.util.KeyUtil
import com.mxt.anitrend.view.fragment.list.MediaBrowseFragment

/**
 * Created by Maxwell on 10/14/2016.
 */
class SeasonPageAdapter(
    fragmentActivity: FragmentActivity,
    context: Context,
) : BaseStatePageAdapter(fragmentActivity, context) {
    init {
        setPagerTitles(R.array.seasons_titles)
    }

    private val fragmentItems by lazy(LAZY_MODE_UNSAFE) {
        arrayOf(KeyUtil.WINTER, KeyUtil.SPRING, KeyUtil.SUMMER, KeyUtil.FALL).map { season ->
            FragmentItem<Fragment>(
                MediaBrowseFragment::class.java,
                Bundle(params).apply {
                    putString(KeyUtil.arg_mediaType, KeyUtil.ANIME)
                    putString(KeyUtil.arg_season, season)
                    putInt(KeyUtil.arg_page_limit, KeyUtil.PAGING_LIMIT)
                },
                "MediaBrowseFragment$season",
            )
        }
    }

    override fun createFragment(position: Int): Fragment = fragmentItems[position].fragmentByTagOrNew(fragmentActivity)
}
